tests/check_consistent.py

```diff
@@ -65,24 +65,22 @@ def check_stubs() -> None:
         allowed = {"METADATA.toml", "README", "README.md", "README.rst", "@tests"}
         assert_consistent_filetypes(dist, kind=".pyi", allowed=allowed)
 
+        tests_dir = dist / "@tests"
+        if tests_dir.exists() and tests_dir.is_dir():
+            py_files_present = bool([file for file in tests_dir.iterdir() if file.suffix == ".py"])
```

Comment:
```suggestion
            py_files_present = any(file for file in tests_dir.iterdir() if file.suffix == ".py")
```
This reads more naturally but has the somewhat undesirable effect of relying on `bool(file)`. Feel free to ignore if you like your version better.

Reply:
I considered doing this -- would you like that more?

```suggestion
            py_files_present = len([file for file in tests_dir.iterdir() if file.suffix == ".py"]) > 0
```

Reply:
Either is fine. I guess the part that bothers me is constructing a list just to check that it's not empty, but it's not worth doing something more complicated.

Reply:
Oh wait, we can just do this, right?

```suggestion
            py_files_present = any(file.suffix == ".py" for file in tests_dir.iterdir())
```